packages/linux/package.mk

```diff
@@ -176,6 +176,34 @@ pre_make_target() {
     sed -i -e "/CONFIG_EXTRA_FIRMWARE_DIR/d" -e "/CONFIG_EXTRA_FIRMWARE=.../a CONFIG_EXTRA_FIRMWARE_DIR=\"external-firmware\"" ${PKG_BUILD}/.config
   fi
 
+  if [ -f "${DISTRO_DIR}/${DISTRO}/required_kernel_options" ]; then
```

Comment:
I guess  `kernel_make oldconfig` should be run before the options are checked, since the Kconfig scripts may change the actual options.

Reply:
I think we should get away from running oldconfig in general as it may change options that we have configured

Reply:
Kconfig scripts will be invoked before the actual kernel built anyway (iirc it does even call the oldconfig target before) - they are just there to ensure options set are sane (and overwrite them if necessary). So: I'd keep it here and just check the options after make oldconfig is called to make sure, we are checking the actual options that are used for kernel build.

Reply:
`make oldconfig` is called right after these changes

Reply:
looks like explained my thoughts wrong :) 
what I tried to say was: Kconf checks/corrections will be done by the kernel target itself (i.e. zImage/Image/uImage/....) anyway - no matter if we call oldconfig before or not. the good thing by calling oldconfig before is,  that we know the result here already  - and can do our checks against it